Source/projections/Store/IProjectionOf.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Cancellation } from '@dolittle/sdk.resilience';
+import { Key } from '../Key';
+import { CurrentState } from './CurrentState';
+import { ScopedProjectionId } from './ScopedProjectionId';
+
+/**
+ * Defines a system that knows about a projection.
+ * @template TReadModel The type of the projection read model.
+ */
+export abstract class IProjectionOf<TReadModel> {
+    /**
+     * Gets the {@link ScopedProjectionId} identifier;.
+     */
+    abstract readonly identifier: ScopedProjectionId;
```

Comment:
I think we might want to split this into `ProjectionId` and `ScopeId` as two properties.